### ISSUE TITLE
Markdown link reference definitions don't work inside paragraphs

### ISSUE THREAD

**Issue by ashleygwilliams** [npm/marky-markdown#159]
via https://github.com/npm/newww/issues/1706 @cianclarke @chojrak11

> https://www.npmjs.com/package/express-session#compatible-session-stores
> It must have worked previously. Related: expressjs/session#248

```
[![★][cassandra-store-image] cassandra-store][cassandra-store-url] An Apache Cassandra-based session store. [cassandra-store-url]: https://www.npmjs.com/package/cassandra-store [cassandra-store-image]: https://img.shields.io/github/stars/webcc/cassandra-store.svg?label=★
```

should give this html
```
<p><a href="https://www.npmjs.com/package/cassandra-store"><img src="https://camo.githubusercontent.com/3e8576e506c9d9aee8274482046d7dd87bbde861/68747470733a2f2f696d672e736869656c64732e696f2f6769746875622f73746172732f77656263632f63617373616e6472612d73746f72652e7376673f6c6162656c3d254532253938253835" alt="★" data-canonical-src="https://img.shields.io/github/stars/webcc/cassandra-store.svg?label=%E2%98%85" style="max-width:100%;"> cassandra-store</a> An Apache Cassandra-based session store.</p>
```

but gives 
```
<p>[![★][cassandra-store-image] cassandra-store][cassandra-store-url] An Apache Cassandra-based session store.
[cassandra-store-url]: <a href="https://www.npmjs.com/package/cassandra-store">https://www.npmjs.com/package/cassandra-store</a>
[cassandra-store-image]: <a href="https://img.shields.io/github/stars/webcc/cassandra-store.svg?label=%E2%98%85">https://img.shields.io/github/stars/webcc/cassandra-store.svg?label=★</a></p>
```

**Comment by ashleygwilliams** [comment]
https://github.com/expressjs/session/issues/248 suggests that the markdown being handed to us is not to spec. that does not mean we shouldn't fix it. the question for us is-- Github parses this, is that a bug or a feature? if a feature, we should fix marky so that this works.

thoughts, @revin?

**Comment by chojrak11** [comment]
It's a feature which makes it easier to create useful documents. IMHO it even warrants a change to Markdown spec if this syntax is not currently supported.

**Comment by revin** [comment]
In general my feeling is that if we *can* figure out how to render something, we should. Like a web browser. I imagine this would require some sort of markdown-it plugin to affect how it handles "link reference definitions".

The [CommonMark spec](http://spec.commonmark.org/0.24/#example-172) specifically says "a link reference definition cannot interrupt a paragraph", so the markdown in that readme is definitely non-spec (and fixing the document is just a matter of adding blank lines between the lines of text and the reference definitions). All the same, it's probably worth looking at supporting it.

**Comment by revin** [comment]
fixed in #164.
